nvflare/tool/code_pre_installer/README.md

```diff
@@ -0,0 +1,286 @@
+# NVFLARE Code Pre-Installer
+
+This tool helps install NVFLARE application code and libraries before running federated learning jobs.
+
+## Overview
+
+The code pre-installer handles:
+- Installation of application code
+- Installation of shared libraries
+- Site-specific customizations
+- Python package dependencies
+
+The tool provides two main commands:
+- `prepare`: Package application code for installation
+- `install`: Install packaged code to target sites
+
+## Directory Structure
+
+Expected application code zip structure:
+
+```
+application.zip
+├── application/<app_name>/
+│               ├── meta.json       # Application metadata
+│               ├── app_<site>/     # Site custom code
+│                  └── custom/      # Site custom code
+├── application-share/              # Shared resources
+│   └── shared.py
+└── requirements.txt       # Python dependencies (optional)
+```
+or
+```
+application.zip
+├── application/<app_name>/
+│               ├── meta.json       # Application metadata
+│               ├── app/            # Site custom code
+│                  └── custom/      # Site custom code
+``` 
+Here is an example of creating a folder structure for pre-installation:
+```bash
+mkdir -p /tmp/nvflare/pre-install/application
+mkdir -p /tmp/nvflare/pre-install/application-share
+```
+
+For example, if the app name is `fedavg`, the directory structure would look like this:
```

Comment:
personally I prefer calling this just job name, since it has exact structure and meaning as our Job.